src/lxc/utils.c

```diff
@@ -1651,13 +1651,29 @@ int run_command(char *buf, size_t buf_size, int (*child_fn)(void *), void *args)
 			buf[bytes - 1] = '\0';
 	}
 
-	fret = wait_for_pid(child);
+	if (wait_status) {
+		fret = lxc_wait_for_pid_status(child);
+	}
+	else {
```

Comment:
Nit:
- don't put a `\n' after a `}` and the `else`
- don't use `{}` around single-line branches